src/policy/policy.cpp

```diff
@@ -271,8 +272,14 @@ bool IsWitnessStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
             // Taproot spend (non-P2SH-wrapped, version 1, witness program size 32; see BIP 341)
             Span stack{tx.vin[i].scriptWitness.stack};
             if (stack.size() >= 2 && !stack.back().empty() && stack.back()[0] == ANNEX_TAG) {
-                // Annexes are nonstandard as long as no semantics are defined for them.
-                return false;
+                const std::vector<unsigned char> &annex = stack.back();
+                // For now allow 0 to MAX_ANNEX_DATA bytes push to allow drop-in replacement via BIP PR#1381
+                // But only one annex per transaction to avoid witness stuffing.
+                if (!allow_annex_data || annex_data_already_found ||
+                    annex.size() < 2 || annex.size() > 1 + 1 + MAX_ANNEX_DATA || (size_t)annex[1] != annex.size() - 2) {
+                    return false;
+                }
+                annex_data_already_found = true;
```

Comment:
Can you walk me through the `annex_bytes_left` starting point plus the "annex_bytes_left -= annex.size() + 1;" part?

In my head:

`annex_bytes_left` would be `MAX_ANNEX_DATA` + `1`(annex marker) + `1`(length of data payload for entry)

and then we'd just do a straight removing of bytes via budget `annex_bytes_left -= annex.size():`

Reply:
If you add ten bytes of annex payload as "helloworld", then you prefix that with two bytes, namely "0x50 0x0A" as the marker and the size, to give you an annex of "0x50 0x0A helloworld" with `annex.size() == 12`. But to include that annex entry in the witness stack, per bip-141 you also need to prefix the actual data by a varint of its size, which is an additional byte. So the additional data to include the annex is actually "0x0C 0x50 0x0A helloworld". Hence the `+3` instead of `+2`, and the `size()+1` instead of just `size()`.

Reply:
done, with a few modifications/comments on top